src/urllib3/util/retry.py

```diff
@@ -461,7 +463,9 @@ def increment(
             if redirect is not None:
                 redirect -= 1
             cause = "too many redirects"
-            redirect_location = response.get_redirect_location()
+            response_redirect_location = response.get_redirect_location()
```

Comment:
Can we remove this if we add a type annotation to `redirect_location` up above like so:

`redirect_location: Optional[str] = None`

Reply:
No, because the return type of `get_redirect_location` is `Union[str, None, Literal[False]]`.
And if we change it to `redirect_location: Union[Optional[str], "Literal[False]"] = None` then we get 
`Argument 5 to "RequestHistory" has incompatible type "Union[str, None, Literal[False]]"; expected "Optional[str]"`
and we need to change `RequestHistory`